tokens_enum.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright: Copyright nsoul97 (Soulounias Nikolaos) 2019.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   nsoul97 (Soulounias Nikolaos)
+ */
+
+#ifndef __TOK_ENUMS_HPP
+#define __TOK_ENUMS_HPP
+
+/*
+Enum class TOK is used to help identify the tokens of DIL language.
+Since this is an enum class, enumerator names are local to the TOK enum.
+Therefore to use an enumerator name we have to use the scope resolution
+operator. e.g. TOK::UNDEFINED
+*/
+enum class TOK {
+  UNDEFINED,   // token of undefined kind (i.e. error)
+  EOI,         // end of input
+  INT_LIT,     // integer literal
+  MAIN,        // main keyword
```

Comment:
`MAIN` is not a special keyword, it's an identifier. The `main` function is just like any other function (with a difference that it has to exist but this does not make its name a special keyword).